crates/subspace-farmer/src/single_disk_farm.rs

```diff
@@ -850,7 +850,7 @@ impl SingleDiskFarm {
         let (start_sender, mut start_receiver) = broadcast::channel::<()>(1);
         let (stop_sender, mut stop_receiver) = broadcast::channel::<()>(1);
         let modifying_sector_index = Arc::<RwLock<Option<SectorIndex>>>::default();
-        let (sectors_to_plot_sender, sectors_to_plot_receiver) = mpsc::channel(0);
+        let (sectors_to_plot_sender, sectors_to_plot_receiver) = mpsc::channel(1);
```

Comment:
It is actually related, without this capacity of the channel will be exactly one message (just one sender), so it would be possible for plotter to not see the future plotted sector even though it is in fact already known by scheduler. By having capacity of `1` here we allow two messages to be "in-flight", ensuring that plotting process can observe both and schedule downloading of the next sector while the previous is still being encoded.

Reply:
I see, it is related but not used right now since we can't use `sectors_to_plot_receiver.try_peek()` to get the next sector index to download it ahead, and instead using `next_segment_index_hint` to get the sector index.

Reply:
True, yes